### ISSUE TITLE
Bug: I cant able to load project

### ISSUE THREAD

**Issue by ircbloqcc** [openblockcc/openblock-desktop#21]
Still some issues in loading project. I cant load the project in OB 2.2.6

![Screenshot (41)](https://user-images.githubusercontent.com/86342573/132447753-b1a1192e-ec42-4801-8147-c0ca97ce63ba.png)

**Comment by zhengyangliu** [comment]
Can you show me your project file? the index.js in the pack.------------------&nbsp;原始邮件&nbsp;------------------
***@***.***&gt;
发送时间:&nbsp;2021年9月8日(星期三) 中午12:43
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[openblockcc/openblock-desktop] Bug: I cant able to load project (#21)

**Comment by ircbloqcc** [comment]
this is the project file

[asdas.zip](https://github.com/openblockcc/openblock-desktop/files/7126313/asdas.zip)

**Comment by zhengyangliu** [comment]
Is this file is  created in the old version？ Its internal deviceextensions configuration is empty, which is due to a bug in the previous version. If you create with a new version, this problem should not exist.

![屏幕截图 2021-09-16 194751](https://user-images.githubusercontent.com/10896145/133606816-59e12999-92ad-4d69-a785-bdbf265c5219.png)

**Comment by zhengyangliu** [comment]
After you confirm, I will close this issue.

**Comment by ircbloqcc** [comment]
> Is this file is created in the old version？ Its internal deviceextensions configuration is empty, which is due to a bug in the previous version. If you create with a new version, this problem should not exist.
> 
> ![屏幕截图 2021-09-16 194751](https://user-images.githubusercontent.com/10896145/133606816-59e12999-92ad-4d69-a785-bdbf265c5219.png)

No i have created with new openblock v2.2.6 only. For conformation vm prerelease is latest only. 

Also i created new and checked again..

![Screenshot 2021-09-16 173238](https://user-images.githubusercontent.com/86342573/133608763-6714b13c-d933-4bd1-a9c4-362b19f99106.png)

**Comment by zhengyangliu** [comment]
I misjudged the problem. Can you show me how to create this bug causing project?

**Comment by ircbloqcc** [comment]
![tst2](https://user-images.githubusercontent.com/86342573/133610928-484993e9-1a89-4e71-9c91-e1f674d843ca.gif)

**Comment by zhengyangliu** [comment]
It's a tough problem. I'll see what's wrong。

**Comment by ircbloqcc** [comment]
I think, problem is with vm only. its not clear the old target, if i edit a old project and saved again means its not clearing old targets. instead of that its adding the stage in side the json file.

![ice_video_20210916-183236_edit_0](https://user-images.githubusercontent.com/86342573/133617697-f6d321cb-cadc-4db6-9f37-68d67c0d389d.gif)

**Comment by zhengyangliu** [comment]
I need to see where the bug is caused.

**Comment by zhengyangliu** [comment]
This bug has been solved in https://github.com/openblockcc/openblock-desktop/commit/17315fd85d6198fcde48ff846becea827ea58692.

**Comment by ircbloqcc** [comment]
Sorry to say, still some issues..
if try to load same project again, code generation area is blank. if i click the costume tab and get back, then code is generating. 
if i load same project, com port is not disconnecting. also upload button not responding. 

![Screenshot 2021-09-18 173013](https://user-images.githubusercontent.com/86342573/133888303-53110d17-e2d9-4cd7-82eb-6f241101fdb9.png)

![Screenshot 2021-09-18 173053](https://user-images.githubusercontent.com/86342573/133888307-57d332ee-0ff4-4b06-b37c-90ea83ac1d59.png)

**Comment by zhengyangliu** [comment]
I can't repeat the first question. I will fix the problem of device disconnection in the next version. At present, you can manually fix this bug by unplugging the USB cable and reconnect the device.

**Comment by zhengyangliu** [comment]
Can you show me how to trigger the first question？

**Comment by ircbloqcc** [comment]
This is the problem

![ice_video_20210918-174404_edit_0(1)](https://user-images.githubusercontent.com/86342573/133888674-e33cdb15-0452-4b06-ae6d-3a33adfdfb0e.gif)

**Comment by zhengyangliu** [comment]
When no deivce extenison is loaded, repeat load this project, the code area will be empty.

**Comment by ircbloqcc** [comment]
Correct.. If any device extensions is loaded, code is generating,,
When no device extension is loaded, code area will be empty.

**Comment by zhengyangliu** [comment]
Fortunately, the bug is not urgent this time, I can fix it a bit slower and relax..

**Comment by ircbloqcc** [comment]
![ice_video_20210918-180746_edit_0](https://user-images.githubusercontent.com/86342573/133889141-42bb0197-7313-41f1-8d6e-a4df1100ea79.gif)

**Comment by ircbloqcc** [comment]
> Fortunately, the bug is not urgent this time, I can fix it a bit slower and relax..

cool..

**Comment by CeemaCR** [comment]
Hi Arthur, Hope you will release the next version soon. Waiting eagerly for a fix of this one....Thanks a lot.

**Comment by zhengyangliu** [comment]
ok, I will update in is week.

**Comment by CeemaCR** [comment]
Thank you so much...

**Comment by CeemaCR** [comment]
Hi Arthur, hope the same change is applicable to the Mac version too

**Comment by zhengyangliu** [comment]
Yes, they are updated synchronously, and they all use the same code.

**Comment by CeemaCR** [comment]
okay thank you . Hope you are working on the Issue with the upload.  And the latest version .7 has given a hint to a lot of new upcoming updates like driver and all...Eagerly waiting to see your updates.

**Comment by zhengyangliu** [comment]
Those bug has been fixed in https://github.com/openblockcc/openblock-desktop/releases/tag/v2.2.8.

**Comment by zhengyangliu** [comment]
I will close this issue after you test the feedback。

**Comment by CeemaCR** [comment]
Thank you so much. Will check and let you know

**Comment by CeemaCR** [comment]
Seems like there is some issue. Can not see any other device other than Arduino  Uno


![1](https://user-images.githubusercontent.com/82566442/136054957-e35f5b37-3d7d-47c5-a3b8-1341b715bdd2.jpeg)

Checked set up file too,  when Arduino Nano is selected in set up file,  app got closed.

**Comment by zhengyangliu** [comment]
The problem was fixed, and I re-released v2.2.8. This bug originated when I updated openblock-link when fixing the problem that microbit v2 could not download the firmware, and some npm packages that openblock-link relied on were updated. These updated packages will cause link crashes.

**Comment by CeemaCR** [comment]
So, will there be a new release for the fix?

**Comment by zhengyangliu** [comment]
Just re-download v2.2.8 directly.

**Comment by CeemaCR** [comment]
Uno only is displayed and that too is not detecting. Same old issue

**Comment by CeemaCR** [comment]
Me only having this issue or anyone else is facing this issue? Any suggestion?

**Comment by CeemaCR** [comment]
Thanks, you Arthur, will check it now..

**Comment by CeemaCR** [comment]
Not sure, whether I should continue here or open a new issue. If I should open a new one, please let me know

As in the video, when we try to upload the code from .ob file, the device is getting disconneted. Any reason for that? And some times the upload takes a lot of time (not .ob file, the direct upload itself) , but, not always


https://user-images.githubusercontent.com/82566442/136644891-87ff7390-2030-45d8-837a-007bff05ef3c.mp4

**Comment by zhengyangliu** [comment]
This is a usage problem. Just click this button when you want to connect the device. If you reload the project, it will automatically cut off the connection with the old device,
![QQ截图20211009131610](https://user-images.githubusercontent.com/10896145/136645089-d63fdef2-6fb7-4920-bb39-4d01dc49b6d9.png)
Click this button to connect the device. I think your previous usage may be a bit misunderstanding. haha.

**Comment by cpyindrajith** [comment]
https://user-images.githubusercontent.com/81285660/136646807-02d2c7f3-169c-4ee5-b334-1e7f15ed9d39.mp4

**Comment by cpyindrajith** [comment]
Showing error alert "Please connect a peripheral first" when trying to upload code from local file. I already connected Arduino UNO device with Openblock

**Comment by ircbloqcc** [comment]
> Showing error alert "Please connect a peripheral first" when trying to upload code from local file. I already connected Arduino UNO device with Openblock

That is not error... Device has been disconnected.. You can reconnect then give upload...

**Comment by zhengyangliu** [comment]
> > Showing error alert "Please connect a peripheral first" when trying to upload code from local file. I already connected Arduino UNO device with Openblock
> 
> That is not error... Device has been disconnected.. You can reconnect then give upload...
> 
> How many times you trying to load a file, device will get disconnect.. So each time you have to connect the com port and give upload..

Right answer.

**Comment by cpyindrajith** [comment]
Do I need to reconnect the device every time I try to upload a saved file?
